src/ui/public/config/config.js

```diff
@@ -129,34 +35,84 @@ module.service('config', function (Private, kbnVersion, kbnIndex, $rootScope, bu
    *                             be stored. Defaults to the config key
    * @return {function} - an unbind function
    */
-  config.$bind = function ($scope, key, property) {
-    if (!property) property = key;
-
-    let update = function () {
-      $scope[property] = config.get(key);
-    };
+  config.$bind = function (scope, key, property = key) {
+    config.on(key, update, scope);
+    function update(value) {
+      scope[property] = value;
+    }
+  };
 
+  config.on = function (key, fn, scope = $rootScope) {
+    const postfix = key === `*` ? `` : `.${key}`;
     update();
-    return _.partial(_.invoke, [
-      $scope.$on('change:config.' + key, update),
-      $scope.$on('init:config', update)
-    ], 'call');
+    scope.$on(`change:config${postfix}`, update);
+    scope.$on(`init:config`, update);
+    function update() {
+      fn(config.get(key));
```

Comment:
This means that `config.on('*', ...)` gets null passed to the handler right? Should `config.get()` be updated to support `*` as well?